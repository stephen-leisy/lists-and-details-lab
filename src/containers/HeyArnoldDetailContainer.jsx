import React from 'react';
import Banner from '../components/header/Banner';
import DetailCharacter from '../components/characters/DetailCharacter';
import { useOneCharacter } from '../hooks/allCharacters';
import Spinner from '../components/characters/Spinner';

export default function HeyArnoldDetailContainer({
  match: {
    params: { id },
  },
}) {
  const { loading, character } = useOneCharacter(id);

  if (loading === true)
    return (
      <main>
        <Banner />
        <Spinner />
        loading...
      </main>
    );
  return (
    <main>
      <Banner />
      <DetailCharacter {...character} />
    </main>
  );
}
