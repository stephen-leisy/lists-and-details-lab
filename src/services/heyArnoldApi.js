export const getCharacters = async () => {
  const response = await fetch(
    `https://hey-arnold-api.herokuapp.com/api/v1/characters`
  );
  const json = await response.json();

  return json.map(({ _id, name, image }) => ({
    name,
    image,
    id: _id,
  }));
};
export const getCharacterById = async (charId) => {
  const response = await fetch(
    `https://hey-arnold-api.herokuapp.com/api/v1/characters/${charId}`
  );
  const { name, image, _id } = await response.json();

  return {
    name,
    image,
    id: _id,
  };
};

export const getRandomGif = async () => {
  const gif = await fetch(
    `https://hey-arnold-api.herokuapp.com/api/v1/gifs/random?count=1`
  );
  const json = await gif.json();
  const { _id, gifLink } = await json[0];

  return {
    id: _id,
    gifLink,
  };
};
