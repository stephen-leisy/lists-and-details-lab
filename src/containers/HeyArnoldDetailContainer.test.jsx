import React from 'react';
import { render, screen } from '@testing-library/react';
import HeyArnoldDetailContainer from './HeyArnoldDetailContainer';
import { MemoryRouter } from 'react-router';
import { getCharacterById } from '../services/heyArnoldApi';

describe('Detail container tests', () => {
  it('renders one character to the screen', async () => {
    render(
      <MemoryRouter>
        <HeyArnoldDetailContainer
          match={{
            params: {
              id: '5da237699734fdcb7bef8f5c',
            },
          }}
        />
      </MemoryRouter>
    );
    screen.getByText('loading...');
    const div = await screen.findAllByRole('list', { name: 'character' });
    expect(div).toMatchSnapshot();
  });
});
