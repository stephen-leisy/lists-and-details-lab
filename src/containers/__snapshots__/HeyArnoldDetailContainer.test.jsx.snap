// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail container tests renders one character to the screen 1`] = `
Array [
  <ul
    aria-label="character"
  >
    <li>
      <h1>
        Nancy
      </h1>
      <img
        alt="picture"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      />
    </li>
  </ul>,
]
`;
